venv/Lib/site-packages/PIL/ImagePalette.py

```
@@ -0,0 +1,286 @@
+            raise ValueError(msg)
+
+    def save(self, fp: str | IO[str]) -> None:
+        """Save palette to text file.
+
+        .. warning:: This method is experimental.
+        """
+        if self.rawmode:
+            msg = "palette contains raw palette data"
+            raise ValueError(msg)
+        if isinstance(fp, str):
+            fp = open(fp, "w")
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)